api/src/services/fetchEntry.ts

```diff
@@ -0,0 +1,19 @@
+import { Knex } from "knex"
+
+interface Entry {
+  'input': string,
+  'output': string,
+  'name': string,
+  'outputUpdated': boolean
+}
+
+export default async (name: string, pg:Knex) => {
+  let result = await pg('graphs')
+  .where({name: name}).select()
+  .then((item:Array<Entry>) => item[0])
+  .catch((err:Error) => {
+    throw err
+  })
+
+  return result || {'input': '', 'output': '', 'name': '', 'outputUpdated': false}
```

Comment:
You can really just return null here instead of empty object.
Just define the return type of this function as (Entry | null).

Reply:
Javascript quite well works with null as it has notation to say that an object can be a null.
Functional languages use Optional class for this.
OOP languages use: https://www.baeldung.com/java-null-object-pattern
but Optional also happens to be implementation of this pattern.